.github/workflows/nightly.yml

```diff
@@ -0,0 +1,41 @@
+name: "Nightly Build"
+on:
+  schedule:
+    - cron: "10 * * * *"
+
+jobs:
+  check-date:
+    runs-on: ubuntu-20.04
+    name: Check latest commit
+    outputs:
+      should_run: ${{ steps.should_run.outputs.should_run }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: print latest_commit
+        run: echo ${{ github.sha }}
+
+      - id: should_run
+        continue-on-error: true
+        name: check latest commit is less than a day
+        if: ${{ github.event_name == 'schedule' }}
+        run: test -z $(git rev-list  --after="24 hours"  ${{ github.sha }}) && echo "::set-output name=should_run::false"
```

Comment:
nifty

Reply:
I have no idea if it actually works. Stack Overflow thumbs are the only evidence.